src/docs/reference/core/KeyVault.Azure/README.md

```diff
@@ -0,0 +1,41 @@
+# Azure Key Vault (`OrchardCore.Azure.KeyVault`)
+
+The Azure Key Vault configuration provider adds app configuration values from the Azure Key Vault in order to safeguard your cryptographic keys and secrets used by your app. It also contains custom override of the DefaultKeyVaultManager class that retrieves secrets from Azure Key Vault and translates ---
+to an underscore (_)  and -- to a colon (:). Both underscores and colons are illegal characters in Azure KeyVault.
+
+Example:
+Key Vault Input: "OrchardCore--OrchardCore---Shells---Database--ConnectionString".
+Output: "OrchardCore:OrchardCore_Shells_Database:ConnectionString".
+See https://github.com/OrchardCMS/OrchardCore/issues/6359.
+
+## Configuration
+You will need to specify the name of your Azure Key Vault and [register a service principal](https://docs.microsoft.com/en-us/azure/key-vault/general/group-permissions-for-apps) in Active Directory for accessing your key vault using an access control policy.
+
+```json
+"OrchardCore_Azure_KeyVault": {
+    "KeyVaultName": "", // Set the name of your Azure Key Vault.
+    "AzureADApplicationId": "", // Set the Azure AD Application Id
```

Comment:
Is that required when using managed identity in Azure? I believe we can also associated an Azure Web App for instance to be allowed to access a KV account without anything to setup.

Reply:
Yes, it's the AzureServiceTokenProvider. I got it working on my OrchardCore project.
Some notes:
I was only able to get in working in a .net standard project (which is fine).
I think there is a new API for .net core, but I couldn't get it to work at the time (6 months ago)